### PR TITLE
Check if a scale type is supported by a channel

### PR DESCRIPTION
**Pull request by kanitw**
Part of #1670

### CODE REVIEW

**Comment by kanitw** [comment]
Wait wait ... wanna fix one more thing

**Comment by kanitw** [comment]
Fixed

File: src/channel.ts [hunk 1]
```diff
@@ -165,3 +174,25 @@ export function getSupportedRole(channel: Channel): SupportedRole {
...
+export function supportScaleType(channel: Channel, scaleType: ScaleType): boolean {
+  switch (channel) {
+    case 'row':
+    case 'column':
+      return scaleType === 'band'; // row / column currently supports band only
+    case 'x':
+    case 'y':
+    case 'size': // TODO: size and opacity can support ordinal with more modification
+    case 'opacity':
+      return !!POSITION_SCALE_TYPE_INDEX[scaleType];
```
Comment (domoritz) [inline src/channel.ts]: `scaleType in POSITION_SCALE_TYPE_INDEX`

File: src/scale.ts [hunk 2]
```diff
@@ -1,42 +1,58 @@
...
-    type === ScaleType.ORDINAL_LOOKUP;
+export type ScaleType = typeof ScaleType.LINEAR |
+  typeof ScaleType.LOG | typeof ScaleType.POW | typeof ScaleType.SQRT |
+  typeof ScaleType.QUANTILE | typeof ScaleType.QUANTIZE | typeof ScaleType.SEQUENTIAL |
+  typeof ScaleType.ORDINAL_LOOKUP | typeof ScaleType.POINT | typeof ScaleType.BAND |
+  typeof ScaleType.TIME | typeof ScaleType.UTC;
+
+export const SCALE_TYPES: ScaleType[] = [
+  // Quantitative
+  'linear', 'log', 'pow', 'sqrt', // TODO: add 'quantile', 'quantize' when we really support them
```
Comment (domoritz) [inline src/scale.ts]: I see you're already using strings constants.

**Pull request merged**